private/Message.hpp

```diff
@@ -15,23 +15,32 @@
 #include <string>
 #include <vector>
 
-class IrcMessage
+class Message
 {
+public:
+	// Constructors
+	Message(std::string const &msg);
+	
+	// Destructor
+	~Message(void);
+
+	// Accessors
+	std::string const        &get_prefix() const;
+	std::string const        &get_command() const;
+	std::vector<std::string> &get_params();
+
+	// Setters
+	void set_prefix(std::string const &prefix);
+	void set_command(std::string const &command);
+	void set_params(std::vector<std::string> const &params);
+
+	// Methods
+	void add_param(std::string const &param);
+
+	void display() const;
```

Comment:
Display being a debug function, it should be removed in the final main.
I do argue that we should keep it for the time being as we do need to test everything by the main fully merged with each of our parts